parastell/invessel_build.py

```diff
@@ -713,27 +713,61 @@ def extract_solids_and_mat_tags(self):
 
         return solids, mat_tags
 
-    def mesh_component_moab(self, component):
-        """Creates a tetrahedral mesh of a single in-vessel component volume
-        via MOAB. This mesh is created using the point cloud of the specified
-        component and as such, the mesh will be one tetrahedron thick.
+    def mesh_components_moab(self, components):
+        """Creates a tetrahedral mesh of in-vessel component volumes via MOAB.
+        This mesh is created using the point cloud of the specified components
+        and as such, each component's mesh will be one tetrahedron thick.
 
         Arguments:
-            component (str): name of in-vessel component to be meshed.
+            components (array of str): array containing the names of the
+                in-vessel components to be meshed.
         """
         self._logger.info(
-            f"Generating tetrahedral mesh of {component} volume via MOAB..."
+            "Generating tetrahedral mesh of in-vessel component(s) via MOAB..."
         )
 
+        def remove_inner_component(component):
+            """Upon identification of a requested component whose meshing is
+            not supported by the MOAB workflow, removes that component from the
+            input list and raises a warning.
+
+            Arguments:
+                component (str): component to be removed.
+            """
+            e = Warning(
+                f"Meshing of {component} volume not supported for MOAB "
+                f"workflow; {component} volume will be removed from list of "
+                "components to be meshed."
+            )
+            self._logger.warning(e.args[0])
+            components.remove(component)
+
+        if "plasma" in components:
+            remove_inner_component("plasma")
+        elif "chamber" in components:
+            remove_inner_component("chamber")
+
         surfaces = []
+        gap_map = []
+
         surface_keys = list(self.Surfaces.keys())
 
-        component_idx = surface_keys.index(component)
-        surfaces = [self.Surfaces[surface_keys[component_idx - 1]]]
-        surfaces.append(self.Surfaces[component])
+        # Identify surfaces and gaps in mesh
+        for component_idx, component in enumerate(components):
+            component_surf_idx = surface_keys.index(component)
+            previous_component = surface_keys[component_surf_idx - 1]
+
+            # Conditionally handle gap in mesh
+            if previous_component != components[component_idx - 1]:
```

Comment:
This assumes that `components` is sorted in the same way that the components were built.  Either error or warning and/or sort them appropriately.